### PR TITLE
Feature/page header as nav link

### PR DESCRIPTION
**Pull request by penx**
Refactor page header to support the `as-nav-link` pattern as detailed at https://github.com/UKHomeOffice/govuk-react/issues/423

However, there is no active style for this component so the main thing that will be used is the `as` property.

screen shot 2018-10-22 at 14 19 44

### CODE REVIEW

File: components/page-header/src/index.js [hunk 1]
```diff
@@ -80,5 +80,5 @@ PageHeader.defaultProps = {
 
 export default PageHeader;
 
-export asLogoAnchor from './hoc/asLogoAnchor';
-export asNavAnchor from './hoc/asNavAnchor';
+export LogoAnchor from './atoms/logo-anchor';
```
Comment (penx) [inline components/page-header/src/index.js]: Export these as static objects on the root component

So they can be used e.g.

```jsx
<PageHeader>
  <PageHeader.NavAnchor>...
```

Comment (penx) [inline components/page-header/src/index.js]: Export these as static objects on the root component

So they can be used e.g.

```jsx
<PageHeader>
  <PageHeader.NavAnchor>...
```

Comment (penx) [inline components/page-header/src/index.js]: Export these as static objects on the root component

So they can be used e.g.

```jsx
<PageHeader>
  <PageHeader.NavAnchor>...
```

Comment (penx) [inline components/page-header/src/index.js]: Export these as static objects on the root component

So they can be used e.g.

```jsx
<PageHeader>
  <PageHeader.NavAnchor>...
```

Comment (penx) [inline components/page-header/src/index.js]: Export these as static objects on the root component

So they can be used e.g.

```jsx
<PageHeader>
  <PageHeader.NavAnchor>...
```

Comment (penx) [inline components/page-header/src/index.js]: Export these as static objects on the root component

So they can be used e.g.

```jsx
<PageHeader>
  <PageHeader.NavAnchor>...
```

File: components/page-header/src/index.js [hunk 2]
```diff
@@ -33,13 +36,13 @@ const NavigationContainer = styled('div')({
...
+ *
  * ```jsx
  * import { Link } from 'react-router-dom';
- * import PageHeader, { asLogoAnchor, asNavAnchor } from '@govuk-frederic/page-header';
- * 
- * const LogoLink = asLogoAnchor(Link);
- * const NavLink = asNavAnchor(Link);
+ * import PageHeader from '@govuk-frederic/page-header';
+ *
+ * const LogoLink = asNavLink()(PageHeader.LogoAnchor);
```
Comment (stevesims) [inline components/page-header/src/index.js]: for this example to work, you'd also need to `import asNavLink from 'as-nav-link';`

File: components/page-header/src/atoms/nav-anchor.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import styled from 'react-emotion';
+import { WHITE, BLACK, YELLOW } from 'govuk-colours';
+import { SPACING } from '@govuk-react/constants';
+
+const NavAnchor = styled(({
+  as: T = 'a',
+  active, // include here so we don't pass to <a />
```
[the remark this answers is not in the document]
Comment (penx) [reply components/page-header/src/atoms/nav-anchor.js]: no, we don't have an active style for this component, we do on page-navigation though

File: components/page-header/src/index.js (hunk 2 above)
Comment (penx) [reply components/page-header/src/index.js]: done, thanks

**Pull request merged**